staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -5348,6 +5348,57 @@ type NodeList struct {
 	Items []Node `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
 
+// +k8s:conversion-gen:explicit-from=net/url.Values
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeLogViewerOptions is the query options for a Node's logs REST call. The options specified here intersect unless
+// called out otherwise.
+type NodeLogViewerOptions struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// SinceTime is an RFC3339 timestamp from which to show logs.
+	// +optional
+	SinceTime *metav1.Time `json:"sinceTime,omitempty" protobuf:"bytes,1,opt,name=sinceTime"`
+
+	// UntilTime is an RFC3339 timestamp until which to show logs.
+	// +optional
+	UntilTime *metav1.Time `json:"untilTime,omitempty" protobuf:"bytes,2,opt,name=untilTime"`
+
+	// TailLines is used to retrieve the specified number of lines (not more than 100k) from the end of the log.
+	// Support for this is implementation specific and only available for Linux and Windows service logs. It will be
+	// ignored for the unsupported case.
+	// +optional
+	TailLines *int64 `json:"tailLines,omitempty" protobuf:"varint,3,opt,name=tailLines"`
+
+	// Format is used to display service logs in an alternate format.
+	// Support for this is implementation specific. Supported only for service logs on Linux. It will be ignored for
+	// the unsupported case. Valid values are: short-precise, json, short, short-unix, short-iso, short-iso-precise,
+	// and cat. It will be ignored for the unsupported case.
+	// TODO: Make this into an enum type
+	// +optional
+	Format string `json:"format,omitempty" protobuf:"bytes,5,opt,name=format"`
+
+	// Pattern filters log entries by the provided regex pattern. On Linux nodes, this pattern will be read as a
+	// PCRE2 regex, on Windows nodes it will be read as a PowerShell regex.
+	// Support for this is implementation specific and only available for Linux and Windows service logs. It will be
+	// ignored for the unsupported case.
+	// +optional
+	Pattern string `json:"pattern,omitempty" protobuf:"bytes,6,opt,name=pattern"`
+
+	// Boot show messages from a specific boot. Allowed values are [-100, 0]. Passing an invalid boot offset will fail
+	// retrieving logs and return an error.
+	// Support for this is implementation specific. Supported only for service logs on Linux. It will be ignored for the
+	// unsupported case.
+	// +optional
+	Boot *int64 `json:"boot,omitempty" protobuf:"varint,8,opt,name=boot"`
+
+	// Query are the specified services(s) or file to return log entries from. The server implements a heuristic to get
+	// the logs from service(s) or a file.  Files are relative to the node's /var/log/ directory. If both service
```

Comment:
"services or files" - once plural always plural

Reply:
"relative to the node's log directory" - we don't neeeeed it to be /var/log literally.